shell/platform/common/cpp/client_wrapper/include/flutter/texture_registrar.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_TEXTURE_REGISTRAR_H_
+
+#include <flutter_texture_registrar.h>
+#include <stdint.h>
+
+#include <memory>
+
+namespace flutter {
+
+// An interface used as an image source by texture widgets.
+class Texture {
+ public:
+  virtual ~Texture() {}
+
+  // Returns a FlutterDesktopPixelBuffer that contains the actual pixel data.
+  // The intended surface size is specified by |width| and |height|.
+  virtual const FlutterDesktopPixelBuffer* CopyPixelBuffer(size_t width,
+                                                           size_t height) = 0;
+};
+
+// An object keeping track of external textures.
+class TextureRegistrar {
+ public:
+  virtual ~TextureRegistrar() {}
+
+  // Registers a |texture| object and returns the ID for that texture.
+  virtual int64_t RegisterTexture(Texture* texture) = 0;
```

Comment:
And then this could take a `std::variant` of texture types, typedef'd to a set list of known texture types. As long as we only added new types to the end of it, that shouldn't break most code when updated (we could explicitly document that code should expect more types to be added, so that people don't do things like write exhaustive visitors of the texture variant—not that I can think of why someone would want to do that anyway).

I don't think there's a good way to do it via a class hierarchy given that the signature of the equivalent to `CopyPixelBuffer` isn't likely to be the same.